x-pack/test/functional/services/ml/api.ts

```diff
@@ -459,6 +459,15 @@ export function MachineLearningAPIProvider({ getService }: FtrProviderContext) {
       return await esSupertest.get(`/_ml/anomaly_detectors/${jobId}`).expect(200);
     },
 
+    async adJobExist(jobId: string) {
```

Comment:
this will always return `true` if a wildcard is used in the job ID.
It'd be better to search for the ID in the returned list, possibly including a regex match if a wildcard is used.
Or reject wildcard uses completely.

It could also be called with a comma separated list of of IDs in a string, which will fail if one is missing.

Reply:
Added job id validation in 2624d61bdc86c605824a40729645f8de47715534 .